include/vcpkg/base/json.h

```diff
@@ -203,22 +209,35 @@ namespace vcpkg::Json
         ~Object() = default;
 
         // asserts if the key is found
-        Value& insert(std::string key, std::string value);
-        Value& insert(std::string key, Value&& value);
-        Value& insert(std::string key, const Value& value);
-        Object& insert(std::string key, Object&& value);
-        Object& insert(std::string key, const Object& value);
-        Array& insert(std::string key, Array&& value);
-        Array& insert(std::string key, const Array& value);
+        template<class StringViewLike,
+                 class = std::enable_if_t<std::is_constructible<StringView, StringViewLike&&>::value>>
+        Value& insert(StringView key, StringViewLike&& value)
+        {
+            return this->insert(key, StringView(static_cast<StringViewLike&&>(value)).to_string());
+        }
+        Value& insert(StringView key, std::string&& value);
```

Comment:
@BillyONeal Calling explicit attention to these lines as ones I adjusted.

Reply:
Proposal:

```
        template<class StringViewLike,
                 class = std::enable_if_t<std::is_constructible<StringView, const StringViewLike&>::value>>
        Value& insert(StringView key, const StringViewLike& value)
```

We are never moving from this input so we can use constT& which is much less psycho greedy.

Reply:
Also, I thought we couldn't use enable_if_t ... guess I should fix the others.